decidim-core/app/assets/javascripts/decidim/vizzs/metrics.js.es6

```diff
@@ -1,26 +1,23 @@
-/* eslint-disable no-unused-vars */
-
-// Outside of the closure to make it public
-let DATACHARTS = null;
-
-const fetchDatacharts = () => {
-
-  const metrics = [{
-    key: "NAME_TO_BE_IN_THE_HTML-1",
-    query: "GRAPHQL_QUERY-1"
-  }, {
-    key: "NAME_TO_BE_IN_THE_HTML-2",
-    query: "GRAPHQL_QUERY-2"
-  }]
-
-  const fetch = (query) => $.post("<-- GRAPHQL_URL -->", query)
-
-  const promises = metrics.map((metric) => fetch(metric.query).then((response) => {
-    DATACHARTS[metric.key] = response
-
-    return DATACHARTS
-  }))
-
-  Promise.all(promises).then(() => DATACHARTS)
-
-}
+/* global areachart */
+
+$(() => {
+  const query = (metric) => {
+    return {query: `{ ${metric} { metric { key value } } }`};
```

Comment:
After a better understanding of this PR, I guess this is where the frontend is asking the backend for the metrics, right?

If so, it would be much better to take advantage of GraphQL and send a single query with all the metrics you need, instead of one query per metric.

Reply:
It can be done, modifying the `query` generation and ajax callback for each metric